server.go

```diff
@@ -740,6 +740,17 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 								negroni.Use(metricsMiddlewareBackend)
 							}
 						}
+
+						if len(frontend.WhitelistSourceRange) > 0 {
+							log.Info("Configured IP Whitelists: ", frontend.WhitelistSourceRange)
+							ipSourceRanges := frontend.WhitelistSourceRange
+							ipWhitelistMiddleware, err := middlewares.NewIPWhitelister(ipSourceRanges)
+							if err != nil {
+								log.Fatal("Error creating IP Whitelister: ", err)
+							}
+							negroni.Use(ipWhitelistMiddleware)
+						}
+
```

Comment:
The Server-Construction-Code is not really nice to be tested, is it? ;)
I extracted the configuration of the middleware into a smaller, easy to test function and tested that. I think this is the better approach instead of configuring a complete Server and trying to fiddle out which middlewares are configured.